yt_dlp/extractor/ximalaya.py

```diff
@@ -201,29 +146,27 @@ def _real_extract(self, url):
                                          note='Download album page for %s' % playlist_id,
                                          errnote='Unable to get album info')
 
-        title = self._html_search_regex(r'detailContent_title[^>]*><h1(?:[^>]+)?>([^<]+)</h1>',
+        title = self._html_search_regex(r'<h1 class="title dC_">([^<]+)</h1>',
                                         webpage, 'title', fatal=False)
 
-        return self.playlist_result(self._entries(webpage, playlist_id, uid), playlist_id, title)
-
-    def _entries(self, page, playlist_id, uid):
-        html = page
-        for page_num in itertools.count(1):
-            for entry in self._process_page(html, uid):
-                yield entry
-
-            next_url = self._search_regex(r'<a\s+href=(["\'])(?P<more>[\S]+)\1[^>]+rel=(["\'])next\3',
-                                          html, 'list_next_url', default=None, group='more')
-            if not next_url:
-                break
-
-            next_full_url = self._BASE_URL_TEMPL % (self.scheme, next_url)
-            html = self._download_webpage(next_full_url, playlist_id)
-
-    def _process_page(self, html, uid):
-        find_from = html.index('album_soundlist')
-        for mobj in re.finditer(self._LIST_VIDEO_RE % uid, html[find_from:]):
-            yield self.url_result(self._BASE_URL_TEMPL % (self.scheme, mobj.group('url')),
-                                  XimalayaIE.ie_key(),
-                                  mobj.group('id'),
-                                  mobj.group('title'))
+        return self.playlist_result(self._entries(playlist_id), playlist_id, title)
+
+    def _entries(self, playlist_id):
+        first_page = self._process_page(playlist_id, 1)
+
+        entries = first_page['tracks']
+
+        max_page_num = math.ceil(first_page['trackTotalCount'] / first_page['pageSize'])
+        for idx in range(2, max_page_num + 1):
+            json = self._process_page(playlist_id, idx)
+            entries += json['tracks']
+
+        return [self.url_result('%s://www.ximalaya.com%s' % (self.scheme, e['url']),
+                XimalayaIE.ie_key(),
+                e['trackId'],
+                e['title'])
+                for e in entries]
+
+    def _process_page(self, playlist_id, page_num):
+        url = f'https://www.ximalaya.com/revision/album/v1/getTracksList?albumId={playlist_id}&pageNum={page_num}&sort=1'
+        return self._download_json(url, playlist_id, note=f'get tracksList page {page_num}')['data']
```

Comment:
feels like it should be a pagedlist

Reply:
done.
it will download first page twice